agent/qrexec-agent.c

```diff
@@ -649,13 +719,17 @@ int wait_for_session_maybe(char *cmdline) {
 
 
 /* hdr parameter is received from dom0, so it is trusted */
-void handle_server_exec_request_init(struct msg_header *hdr)
+static void handle_server_exec_request_init(struct msg_header *hdr)
 {
     struct exec_params params;
     int buf_len = hdr->len-sizeof(params);
-    char buf[buf_len];
+    char *buf = malloc(buf_len);
+    if (!buf) abort();
 
-    assert(hdr->len >= sizeof(params));
+#if defined buf || defined len || defined sizeof || !defined assert || defined params
+#error bug
+#endif
```

Comment:
What exactly are you trying to do here? Combat some "clever" malicious header file? Please don't, it only obfuscate the code. You really can assume same basic sanity of the build environment (otherwise all is lost anyway).